google-cloud-resourcemanager/src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/HttpResourceManagerRpc.java

```diff
@@ -16,58 +16,168 @@
 import com.google.cloud.Tuple;
+import com.google.cloud.http.BaseHttpServiceException;
 import com.google.cloud.http.HttpTransportOptions;
 import com.google.cloud.resourcemanager.ResourceManagerException;
 import com.google.cloud.resourcemanager.ResourceManagerOptions;
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
 import java.io.IOException;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CancellationException;
+import org.threeten.bp.Duration;
 
 public class HttpResourceManagerRpc implements ResourceManagerRpc {
 
-  private final Cloudresourcemanager resourceManager;
+  private static final JsonFactory JSON_FACTORY =
+      new JacksonFactory();
+
+  // See doc of create() for more details:
+  // https://developers.google.com/resources/api-libraries/documentation/cloudresourcemanager/v1/java/latest/com/google/api/services/cloudresourcemanager/CloudResourceManager.Projects.html#create(com.google.api.services.cloudresourcemanager.model.Project)
+  private static final RetrySettings CREATE_RETRY_SETTINGS =
+      RetrySettings.newBuilder()
+          // SLO permits 30s at 90th percentile, double it for total limit
+          .setTotalTimeout(Duration.ofMinutes(1))
+          .setInitialRetryDelay(Duration.ofSeconds(5))
+          .setRetryDelayMultiplier(1.5)
+          // Observed P95 latency is 11s. We probably shouldn't sleep longer than this.
+          .setMaxRetryDelay(Duration.ofSeconds(11))
+          .setJittered(false)
+          .setInitialRpcTimeout(Duration.ofSeconds(5))
+          .setMaxRpcTimeout(Duration.ofSeconds(5))
+          .build();
+
+  // reference: https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto
+  private static final ImmutableMap<Integer, Integer> RPC_TO_HTTP_CODES =
+      ImmutableMap.<Integer, Integer>builder()
+          .put(0, 200)
+          .put(1, 499)
+          .put(2, 500)
+          .put(3, 400)
+          .put(4, 504)
+          .put(5, 404)
+          .put(6, 409)
+          .put(7, 403)
+          .put(16, 401)
+          .put(8, 429)
+          .put(9, 400)
+          .put(10, 409)
+          .put(11, 400)
+          .put(12, 501)
+          .put(13, 500)
+          .put(14, 503)
+          .put(15, 500)
+          .build();
+
+  private static final ResultRetryAlgorithm<Operation> OPERATION_HANDLER =
+      new ResultRetryAlgorithm<Operation>() {
+        @Override
+        public TimedAttemptSettings createNextAttempt(
+            Throwable prevThrowable, Operation prevResponse, TimedAttemptSettings prevSettings) {
+          return null;
+        }
+
+        @Override
+        public boolean shouldRetry(Throwable prevThrowable, Operation prevOp)
+            throws CancellationException {
+          return prevThrowable != null || prevOp.getDone() == null || !prevOp.getDone();
```

Comment:
`prevThrowable != null` looks a bit suspicious, as it means "keep retrying no matter what". Usually there are some retriable and not retriable exceptions/error codes. Whould we really retry on any throwable here?

Reply:
Done.